packages/signer/signer-btc/src/internal/app-binder/task/SendSignMessageTask.test.ts

```diff
@@ -0,0 +1,330 @@
+import {
+  CommandResultFactory,
+  CommandResultStatus,
+  InvalidStatusWordError,
+} from "@ledgerhq/device-management-kit";
+import { type ApduResponse } from "@ledgerhq/device-management-kit";
+import { makeDeviceActionInternalApiMock } from "@ledgerhq/device-management-kit/src/api/device-action/__test-utils__/makeInternalApi.js";
+import { Left, Right } from "purify-ts";
+
+import { type Signature } from "@api/model/Signature";
+import { ClientCommandHandlerError } from "@internal/app-binder/command/client-command-handlers/Errors";
+import { ClientCommandInterpreter } from "@internal/app-binder/command/service/ClientCommandInterpreter";
+import {
+  BUFFER_SIZE,
+  CHUNK_SIZE,
+  ClientCommandCodes,
+  SW_INTERRUPTED_EXECUTION,
+} from "@internal/app-binder/command/utils/constants";
+import { DefaultDataStoreService } from "@internal/data-store/service/DefaultDataStoreService";
+
+import { SendSignMessageTask } from "./SignMessageTask";
+
+const EXACT_ONE_CHUNK_MESSAGE = "a".repeat(CHUNK_SIZE);
+const EXACT_TWO_CHUNKS_MESSAGE = "a".repeat(CHUNK_SIZE * 2);
+const DERIVATION_PATH = "44'/0'/0'/0/0";
+const PREIMAGE = new Uint8Array([1, 2, 3, 4]);
+const MERKLE_ROOT = new Uint8Array(BUFFER_SIZE).fill(0x01);
+
+const SIGNATURE: Signature = {
+  v: 27,
+  r: "0x1212121212121212121212121212121212121212121212121212121212121212",
+  s: "0x3434343434343434343434343434343434343434343434343434343434343434",
+};
+
+const APDU_RESPONSE_YELD: ApduResponse = {
+  statusCode: SW_INTERRUPTED_EXECUTION,
+  data: new Uint8Array([ClientCommandCodes.YIELD]),
+};
+
+describe("SendSignMessageTask", () => {
+  const signatureResult = CommandResultFactory<Signature, Error>({
+    data: SIGNATURE,
+  });
+  const apiMock = makeDeviceActionInternalApiMock();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("run", () => {
+    it("should correctly chunk a message that fits in 1 chunk", async () => {
+      // GIVEN
+      const args = {
+        derivationPath: DERIVATION_PATH,
+        message: EXACT_ONE_CHUNK_MESSAGE,
+      };
+
+      jest
+        .spyOn(DefaultDataStoreService.prototype, "merklizeChunks")
+        .mockImplementation((_, chunks) => {
+          expect(chunks.length).toBe(1);
+          return MERKLE_ROOT;
+        });
+
+      apiMock.sendCommand.mockResolvedValueOnce(signatureResult);
+
+      // WHEN
+      const result = await new SendSignMessageTask(apiMock, args).run();
+
+      // THEN
+      expect(apiMock.sendCommand).toHaveBeenCalledTimes(1);
+      expect(result.status).toBe(CommandResultStatus.Success);
+      if (result.status === CommandResultStatus.Success) {
```

Comment:
each condition should be tested in a test, with this expect you don't have to verify the other case: 

```suggestion
     expect(result).toStrictEqual(CommandResultFactory({ data: SIGNATURE }));
```